isis/src/mex/apps/hrsc2isis/main.cpp

```diff
@@ -11,93 +11,154 @@
+           (label["DETECTOR_ID"][0] == "MEX_HRSC_NADIR") ||
+           (label["DETECTOR_ID"][0] == "MEX_HRSC_GREEN") ||
+           (label["DETECTOR_ID"][0] == "MEX_HRSC_P1") ||
+           (label["DETECTOR_ID"][0] == "MEX_HRSC_IR") ||
+           (label["DETECTOR_ID"][0] == "MEX_HRSC_S1")) {
+    isSrcFile = false;
   }
-  catch(IException &e) {
-    IString msg = "File [" + ui.GetFileName("FROM") +
-                  "] does not appear to be a Mars Express HRSC image.";
+  else {
+    QString msg = "File [" + ui.GetFileName("FROM");
+    msg += "] does not appear to be a Mars Express stereo or SRC file. Label keyword [DETECTOR_ID = ";
+    msg += label["DETECTOR_ID"][0] + "] is not recognized.";
     throw IException(IException::User, msg, _FILEINFO_);
   }
 
-  ProcessImportPds p;
-  Pvl label;
-  lineInFile.clear();
-  numLinesSkipped = 0;
+  // This progream is setup to work with Mex HRSC processing level 1 and 2 only.
+  // Not level 3 (Mapped)
+  if ((int)label["PROCESSING_LEVEL_ID"] >= 3) {
+    QString msg = "File [" + ui.GetFileName("FROM");
+    msg += "] has keyword [PROCESSING_LEVEL_ID = " + label["PROCESSING_LEVEL_ID"][0] + "]";
+    msg += " and can not be read by this program.";
+    throw IException(IException::User, msg, _FILEINFO_);
+  }
 
-  p.SetPdsFile(ui.GetFileName("FROM"), "", label);
+  // The processing for Stereo and SRC are significantly different. Call the
+  // appropriate processing function
+  if (isSrcFile) {
+    ImportHrscSrcImage(p, label);
+  }
+  else {
+    ImportHrscStereoImage(p, label);
+  }
 
-  CubeAttributeOutput outAtt(ui.GetFileName("TO"));
-  outCube = new Cube();
 
-  outCube->setByteOrder(outAtt.byteOrder());
-  outCube->setFormat(outAtt.fileFormat());
-  outCube->setLabelsAttached(outAtt.labelAttachment() == AttachedLabel);
+}
 
-  /**
-    * Isis2 mex2isis.pl:
-    *   if (index($detector_id,"MEX_HRSC_SRC") < 0 &&
-    *   $processing_level_id < 3)
-    */
 
-  bool isSrcFile = (label["DETECTOR_ID"][0] != "MEX_HRSC_SRC");
-  bool mapProjRdr = ((int)label["PROCESSING_LEVEL_ID"] >= 3);
+// Import a PDS3, HRSC, SRC Camera image.
+void ImportHrscSrcImage(ProcessImportPds &p, Pvl &label) {
 
-  try {
-    if (!isSrcFile) {
-      QString msg = "File [" + ui.GetFileName("FROM");
-      msg += "] is SRC data and cannot be read.";
-      throw IException(IException::User, msg, _FILEINFO_);
-    }
+  UserInterface &ui = Application::GetUserInterface();
 
-    if (mapProjRdr) {
-      QString msg = "File [" + ui.GetFileName("FROM");
-      msg += "] is map projected and cannot be read.";
-      throw IException(IException::User, msg, _FILEINFO_);
-    }
+  outCube = p.SetOutputCube("TO");
+  p.StartProcess();
+
+  Pvl otherLabels;
+  TranslateHrscLabels(label, otherLabels);
+
+  std::cout << "========================================" <<
+    otherLabels << "=======================================================" << std::endl;
+
+  if (otherLabels.hasGroup("Instrument") &&
```

Comment:
Any reason to not just write out the whole translated label at once?

Reply:
This was the way the previous version did it, so just didn't change it